src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -203,7 +207,10 @@ private async Task ProcessQueueAsync()
                         await work.CallbackAsync(context, cancellationToken).ConfigureAwait(false);
 
                         // Now that we've mutated our solution, clear out our saved state to ensure it gets recalculated
-                        _lspSolutionCache.Remove(context.Solution.Workspace);
+                        // If the handler doesn't use a solution we just assume the first workspace, which matches the
+                        // assumptions in RequestContext.Create if a request doens't have a document
+                        var workspace = work.RequiresLSPSolution ? context.Solution!.Workspace : _workspaceRegistrationService.GetAllRegistrations().First();
```

Comment:
You're right, this should be fixed. Interestingly this means we don't need the extra logging you just added today, because we need a non-null return from GetDocumentIdentifier for the mutating requests 😛